File: src/main/java/br/com/blz/testjava/data/response/ProductResponse.java
package br.com.blz.testjava.data.response;


import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Builder;
import lombok.Data;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Builder
public class ProductResponse {


    private String sku;
    private String name;
    private InventoryResponse inventory;
    private boolean isMarketable;
}

File: src/main/kotlin/com/boticario/testkotlin/application/response/ProductResponse.kt
package com.boticario.testkotlin.application.response

import com.boticario.testkotlin.domain.entity.Inventory
import com.boticario.testkotlin.domain.entity.Product
import java.io.InvalidObjectException

data class ProductResponse(
        val sku: Long,
        val name: String,
        val inventory: InventoryResponse,
        val isMarketable: Boolean
) {
  companion object {
    fun fromDomain(product: Product) = ProductResponse(
      product.sku,
      product.name,
      InventoryResponse.fromDomain(product.inventory),
      product.isMarketable)
  }
}
